marklogic-client-api/src/test/java/com/marklogic/client/impl/OpticPlanSearchOptionsTest.java

```diff
@@ -0,0 +1,53 @@
+package com.marklogic.client.impl;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.row.RowManager;
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.row.RowSet;
+import com.marklogic.client.test.Common;
+import com.marklogic.client.test.junit5.RequiresML12;
+import com.marklogic.client.type.PlanSearchOptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(RequiresML12.class)
+class OpticPlanSearchOptionsTest {
+	private DatabaseClient client;
+	private RowManager rowManager;
+	private PlanBuilder op;
+
+	@BeforeEach
+	void beforeEach() {
+		client = Common.newClient();
+		rowManager = client.newRowManager();
+		op = rowManager.newPlanBuilder();
+	}
+
+	@Test
+	void bm25() {
+		PlanBuilderBaseImpl.PlanSearchOptionsImpl options =
```

Comment:
We don't want users to have to instantiate these classes themselves (generally, we don't want a user interacting with any class with "Impl" in the name). There's a `PlanBuilderBase.searchOptions()` method (that is unfortunately not used by any tests yet) that the user should call instead.